src/io/flutter/view/InspectorPanel.java

```diff
@@ -538,16 +543,39 @@ public void showProperties(DiagnosticsNode diagnostic) {
           return;
         }
 
-        properties.sort(Comparator.comparing(DiagnosticsNode::getName));
+        if (properties.size() == 0) {
+          // TODO(jacobr): display message that there are no properties
+          // and that no errors occurred.
+          return;
+        }
 
-        final ListTreeTableModelOnColumns model = getTreeModel();
-        final DefaultMutableTreeNode root = new DefaultMutableTreeNode();
+        // Preload all information we need about each property before instantiating
+        // the UI so that the property display UI does not have to deal with values
+        // that are not yet available. As the number of properties is small this is
+        // a reasonable tradeoff.
+        final CompletableFuture[] futures = new CompletableFuture[properties.size()];
+        int i = 0;
         for (DiagnosticsNode property : properties) {
-          if (property.getLevel() != DiagnosticLevel.hidden) {
-            root.add(new DefaultMutableTreeNode(property));
-          }
+          futures[i] = property.getValueProperties();
+          ++i;
         }
-        model.setRoot(root);
+        whenCompleteUiThread(CompletableFuture.allOf(futures), (Void ignored, Throwable errorGettingInstances) -> {
+          if (errorGettingInstances != null) {
+            // TODO(jacobr): show error message explaining properties could not
+            // be loaded.
+            LOG.error(errorGettingInstances);
```

Comment:
Could also use the centered display for this (w/ different text)?

Reply:
done